src/apphosting/secrets/dialogs.ts

```diff
@@ -39,7 +43,10 @@ export function toMetadata(
 }
 
 const matchesServiceAccounts = (target: BackendMetadata) => (test: BackendMetadata) => {
-  return target.serviceAccounts.length === test.serviceAccounts.length && target.serviceAccounts.every(sa => test.serviceAccounts.indexOf(sa) != -1);
+  return (
+    target.serviceAccounts.length === test.serviceAccounts.length &&
+    target.serviceAccounts.every((sa) => test.serviceAccounts.indexOf(sa) !== -1)
```

Comment:
Having BackendMetadata.serviceAccounts be a `Set` could help making equality checks a little more straightforward. Otherwise, we end up with an order-dependent equality check which _may_ be unexpected